crates/ruff_macros/src/map_codes.rs

```diff
@@ -352,8 +351,11 @@ fn generate_iter_impl(
         }
 
         impl RuleCodePrefix {
-            pub fn iter() -> ::std::vec::IntoIter<RuleCodePrefix> {
-                vec![ #(#all_codes,)* ].into_iter()
+            pub fn iter() -> impl Iterator<Item = RuleCodePrefix> {
+                use strum::IntoEnumIterator;
+
+                std::iter::empty()
+                    #(.chain(#linter_idents::iter().map(|x| Self::#linter_idents(x))))*
```

Comment:
Why do we need the `std::iter::empty` and chain?

Reply:
Putting every linter in a slice generates a type error (the map contains the type of the linter group); I used `iter::empty()` because afaik quote has no utils for special casing the first entry entry in an iteration.